src/Conversion/ONNXToTOSA/ONNXToTOSALegalizeUtils.cpp

```diff
@@ -30,6 +30,25 @@
 namespace mlir {
 namespace tosa {
 
+Value createTosaTransposedTensor(PatternRewriter &rewriter, Operation *op,
+    Value &value, llvm::ArrayRef<int64_t> perm) {
+  // Create Permutation Attr
+  DenseElementsAttr permAttr = DenseIntElementsAttr::get(
+      RankedTensorType::get({value.getType().cast<TensorType>().getRank()},
+          rewriter.getI64Type()),
+      perm);
+  // Create Permutation Const
+  Value permList = tosa::CreateOpAndInfer<tosa::ConstOp>(
+      rewriter, op->getLoc(), permAttr.getType(), permAttr);
```

Comment:
You can use `getConstTensor` that already does this operation but with a little bit more checking